### PR TITLE
Add Rangy Class Applier module to Rangy

### PR DESCRIPTION
**Pull request by MaximeKjaer**
This PR introduces type definitions for the [Rangy Class Applier module](https://github.com/timdown/rangy/wiki/Class-Applier-Module) which is a part of [Rangy](https://github.com/timdown/rangy).

This module is a global plugin, since it adds a method to the global `RangyStatic` object `rangy`, and adds its own `ClassApplier` type to the global scope. As such I've tried to follow the [global plugin template](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/global-plugin-d-ts.html) and to add annotations for the dependence on the [global `rangy-core` library](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#dependencies-on-global-libraries). But this is my first contribution to DefinitelyTyped, so please tell me if that isn't the way to go.

***

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/timdown/rangy/wiki/Class-Applier-Module
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by MaximeKjaer** [comment]
@MidnightDesign @ashubham

File: types/rangy/rangy-classapplier.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Type definitions for Rangy Class Applier module
+// Project: https://github.com/timdown/rangy
+// Definitions by: Maxime Kjaer <https://github.com/MaximeKjaer>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+///<reference path="index.d.ts"/>
+
+interface RangyStatic {
+    createClassApplier(theClass: string, options?: RangyClassApplierOptions, tagNames?: Array<string> | string): RangyClassApplier;
```
Comment (MidnightDesign) [inline types/rangy/rangy-classapplier.d.ts]: Why not `string[]`?

File: types/rangy/rangy-classapplier.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+interface RangyClassApplierOptions {
+    elementTagName?: string;
+    elementProperties?: {[property: string]: any};
+    elementAttributes?: {[attribute: string]: any};
```
Comment (MidnightDesign) [inline types/rangy/rangy-classapplier.d.ts]: `any`? Can't this be more restrictive?

File: types/rangy/rangy-classapplier.d.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ignoreWhiteSpace?: boolean;
+    applyToEditableOnly?: boolean;
+    tagNames?: Array<string> | string;
+    normalize?: boolean;
+    onElementCreate?: (element: Element, classApplier: RangyClassApplier) => void;
+    useExistingElements?: boolean;
+}
+
+interface RangyClassApplier extends RangyClassApplierOptions {
```
Comment (MidnightDesign) [inline types/rangy/rangy-classapplier.d.ts]: What's up with all this returned anys? What gets returned there?

**Review by MidnightDesign — CHANGES_REQUESTED** [review]

**Review by MaximeKjaer — COMMENTED** [review]
Added @MidnightDesign's suggestions

File: types/rangy/rangy-classapplier.d.ts (hunk 3 above)
Comment (MidnightDesign) [inline types/rangy/rangy-classapplier.d.ts]: The return type of `getNativeTextRange()` seems to be [well defined](https://github.com/timdown/rangy/wiki/Rangy-Selection#getnativetextrange) for example.

**Review by MidnightDesign — APPROVED** [review]

**Comment by MidnightDesign** [comment]
LGTM. Thank you!

File: types/rangy/rangy-classapplier.d.ts (hunk 3 above)
Comment (MaximeKjaer) [inline types/rangy/rangy-classapplier.d.ts]: Yeah that should be tightened up a bit. But `getNativeTextRange` isn't part of the Class Applier module, so I'll leave that for a separate PR.

My intention here was to do a first PR of a manageable size to get a feel for the whole process of submitting to DefinitelyTyped, and to do a bigger PR solving all the issues with Rangy after that (there are some other modules that would need to be typed, and some methods like `rangy.init()` are still undefined here). I'm happy to take care of them, just in the next PR 😊

**Pull request merged**